src/form/time/dropdown/util/timeDropdownUtils.ts

```diff
@@ -0,0 +1,107 @@
+import parse from "date-fns/parse";
+import addMinutes from "date-fns/addMinutes";
+import formatDate from "date-fns/format";
+import startOfToday from "date-fns/startOfToday";
+
+import {truncateDecimalPart} from "../../../../core/utils/number/numberUtils";
+import {
+  DATE_FORMAT,
+  DAY_IN_S,
+  MINUTE_IN_S,
+  HOUR_IN_MINS
+} from "../../../../core/utils/time/timeConstants";
+import {formatTimeStringTo12hFormatWithMeridiem} from "../../../../core/utils/time/timeUtils";
+import {TIME_DROPDOWN_OPTION_INTERVALS} from "./timeDropdownConstants";
+import {TimeDropdownOption} from "../TimeDropdown";
+
+function generateTimeDropdownOptionFromStartTimeDate(
+  startTimeDate: Date,
+  differenceInMinutes: number
+): TimeDropdownOption {
+  const date = addMinutes(startTimeDate, differenceInMinutes);
+  const timeString = formatDate(date, DATE_FORMAT.LONG_TIME_FORMAT);
+
+  return {
+    id: timeString,
+    title: timeString,
+    subtitle: generateTimeDropdownOptionSubtitle(differenceInMinutes),
+    context: {
+      date,
+      differenceInMinutes
+    }
+  };
+}
+
+function generateTimeDropdownOptionsFromStartTimeDate(
+  startTimeDate: Date,
+  {
+    arrayLength,
+    interval = TIME_DROPDOWN_OPTION_INTERVALS.DEFAULT,
+    startTimeOffsetInMins = 0
+  }: {arrayLength: number; interval?: number; startTimeOffsetInMins?: number}
+) {
+  return new Array(arrayLength)
+    .fill(0)
+    .map<TimeDropdownOption>((_item, index) =>
+      generateTimeDropdownOptionFromStartTimeDate(
+        startTimeDate,
+        startTimeOffsetInMins + index * interval
+      )
+    );
+}
+
+function generateTimeDropdownOptions(options?: {
+  startDate?: Date | null;
+  startTime?: string;
+}): TimeDropdownOption[] {
+  let dropdownOptions: TimeDropdownOption[] = [];
+  const {startDate, startTime} = options || {};
+
+  if (startTime) {
+    const startTimeDate = parse(
+      formatTimeStringTo12hFormatWithMeridiem(startTime),
+      DATE_FORMAT.LONG_TIME_FORMAT,
+      startDate || new Date()
+    );
+
+    // First four options are separated by only 15 mins
+    const firstHourOptions = generateTimeDropdownOptionsFromStartTimeDate(startTimeDate, {
+      // eslint-disable-next-line no-magic-numbers
+      arrayLength: Math.floor(HOUR_IN_MINS / TIME_DROPDOWN_OPTION_INTERVALS.FIRST_HOUR),
```

Comment:
It would be great if we can get the `TIME_DROPDOWN_OPTION_INTERVALS` object dynamically from the params